drivers/temperature/adt7420/adt7420.c

```diff
@@ -347,37 +366,29 @@ float adt7420_get_temperature(struct adt7420_dev *dev)
 }
 
 
-/**************************************************************************//**
- * @brief Write to input shift register SPI interface.
+/***************************************************************************//**
+ * @brief Reads the value of a register SPI interface device.
  *
- * @param dev               - The device structure.
- * @param register_address  - Command control bits.
- * @param data              - Data to be written in input register.
+ * @param dev              - The device structure.
+ * @param register_address - Address of the register.
  *
- * @return  read_back_data  - value read from register.
-******************************************************************************/
-uint16_t set_shift_reg(struct adt7420_dev *dev,
-		       uint8_t register_address,
-		       uint8_t *data)
+ * @return register_value  - Value of the register.
+*******************************************************************************/
+uint16_t adt7420_spi_get_register_value(struct adt7420_dev *dev,
```

Comment:
rename this also to `adt7420_reg_read` to be alligned with the rest for the functions, pass a pointer where to store the data, and return error code instead.